packages/emeralt-database-mongodb/src/index.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  CEmeraltDatabase,
+  IEmeraltDatabase,
+  TVersion,
+  TMetadata,
+} from '@emeralt/types'
+import { MongoClient, Db, Collection } from 'mongodb'
+
+type Options = {
+  url: string
+  dbName: string
+}
+
+class CEmeraltDatabaseMongoDB implements CEmeraltDatabase {
+  storage: {
+    metadatas: Collection
+    versions: Collection
+  }
+
+  constructor(private db: Db) {
+    this.storage = {
+      metadatas: this.db.collection('metadatas'),
+      versions: this.db.collection('versions'),
+    }
+  }
+
+  public getMetadatas() {
+    return this.storage.metadatas
+      .find({}, { projection: { _id: false } })
+      .toArray()
+      .then((metadatas) =>
+        metadatas.reduce(
+          (acc, curr) => ({
+            ...acc,
+            [curr.name]: curr,
+          }),
+          {},
+        ),
+      )
+  }
+
+  public hasMetadata(name: string) {
+    return this.getMetadata(name).then((metadata) => Boolean(metadata))
```

Comment:
`hasMetadata` is one of the most performance impacting methods in database plugin. MongoDB has a way to check for document existence without actually getting its data, which is faster.